Revert "Make torch_geometric models compatible with export (#123403)"

chunyuan-w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128377

This reverts commit d78991a7381adb3df5e9b63c365db4506643edce.

This PR reverts https://github.com/pytorch/pytorch/pull/123403 to fix the performance regression as discussed in https://github.com/pytorch/pytorch/issues/127513#issuecomment-2158835653.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

chunyuan-w committed:
Revert "Make torch_geometric models compatible with export (#123403)"

This reverts commit d78991a7381adb3df5e9b63c365db4506643edce.

[ghstack-poisoned]

chunyuan-w committed:
Update on "Revert "Make torch_geometric models compatible with export (#123403)""


This reverts commit d78991a7381adb3df5e9b63c365db4506643edce.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

angelayi commented on benchmarks/dynamo/torchbench.py:
```diff
@@ -26,20 +26,6 @@
-
-    model._register_state_dict_hook(state_dict_hook)
-
-
```
I think we can still keep this change?

chunyuan-w replied:
I've added that back and only changed the export related part

chunyuan-w committed:
Update on "Revert "Make torch_geometric models compatible with export (#123403)""


This reverts commit d78991a7381adb3df5e9b63c365db4506643edce.

This PR reverts https://github.com/pytorch/pytorch/pull/123403 to fix the performance regression as discussed in https://github.com/pytorch/pytorch/issues/127513#issuecomment-2158835653.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

desertfire commented:
Is the idea to cherry-pick this to the 2.4 branch?

chunyuan-w commented:
> Is the idea to cherry-pick this to the 2.4 branch?

Yes, I plan to submit a cherry-pick PR to 2.4 when the current PR lands to main.

chunyuan-w commented:
@pytorchbot rebase

chunyuan-w committed:
Update

[ghstack-poisoned]

chunyuan-w commented:
@pytorchbot merge
